base/error.jl

```diff
@@ -52,28 +52,31 @@ end
 
 
 """
-    retry(f, [condition]; n=3; max_delay=10) -> Function
```

Comment:
Unless you have a compelling use-case where the new defaults result in superior performance,
I would suggest setting `n == 2` by default and leaving `max_delay=10` as is.

I understand that `2` and `10` might seems like arbitrary numbers at first glance, but I assure you that they were not arrived at by accident.

As far as I understand best practice for retry loops, the following should be observed when choosing default settings:
 - It is almost never a good idea to retry without any delay at all. A small sleep  (`retry` uses ~50ms on 2nd attempt) reduces the odds that the failure cause is still present and gives other tasks a chance to execute (perhaps resolving the failure cause). 50ms should be small enough that the delay is negligible in the cases where time is not a factor in resolving the failure cause.
 - Many failure causes are almost immediately self-resolving (e.g. temporary network saturation, or random transmission corruption). It is therefore a good idea make the 2nd attempt after only a short delay (e.g. ~50ms).
 - Many failure causes are not immediately self-resolving but are resolved without intervention on the order of 100s of ms. e.g. the result of the pervious operation is not yet visible in a distributed system due to a small data replication delay. It therefore makes sense to delay a bit longer if the 2nd attempt didn't work. ( `retry` delays ~250ms on 3rd attempt)
 - It should be assumed that multiple competitors for shared resources all use the same retry strategy. Therefore that strategy should include random jitter of retry delays to reduce the odds of simultaneous retries.

I would argue that `n == 2` is a sensible default because it results in: a 2nd attempt after a very short delay and a 3rd attempt after a 1/4 second delay. This should cover a good proportion of failure causes and provides a balance between retrying quickly and allowing time for a failure to be resolved.

In the case where the user really wants to wait for success, and sets, e.g. `n = 100` it is generally considered poor form to retry too frequently. Normal best practice is to implement exponential back-off to avoid the situation where a temporary overload causes a storm of retries; and the effect of the storm is worse than the original overload. `retry` uses: 50ms, 250ms, 1250ms, 6250ms ... However, at some point the exponential increase gets silly, so a cap is needed. `retry` uses a default cap of 10s. `max_delay=10`.

Setting `max_delay=0` and attempting to access a busy remote resource 100 times would result in the request being repeated as quickly as possible. The likely result is that the server spends more time telling people that it is busy than it does getting on with doing work.

If we were going to tune these parameters, I think `n == 3` and `max_delay=30` would probably give good results in a broad range of use cases, but I don't see value in setting `n` to less than `2`, or `max_delay` to less than `10`.